### PR TITLE
feat(eslint): add inclusive-language rule

### PR DESCRIPTION
**Pull request by smackfu**
## Description

Adds the "inclusive-language/use-inclusive-words" rule:
https://github.com/muenzpraeger/eslint-plugin-inclusive-language

The list of words is explicit to reduce the chance of unintended changes from the upstream dependency.

The rule is currently marked as a warning. Not sure if an error would be warranted? 

Please make sure that the PR fulfills these requirements

- [x] I checked that there aren't other open Pull Requests for the same update/change.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] Rule changes includes a comment to describe the reasoning behind the change.
- [x] PR contains a single rule change.
    
## Motivation and Context

Using inclusive coding terms is essential to creating an inclusive environment. And anything that can be automatically tested by eslint should be.

## How Has This Been Tested?

Ran `npm pack` and installed the repo against an existing repository that was using the existing lint config.  After adding a violation to the code, running eslint generated the expected warning:

```
  10:3   warning  The usage of the non-inclusive word 'master' is discouraged, use 'main' instead  inclusive-language/use-inclusive-words
```

Also, you can see the suggestions in WebStorm here:

Screen Shot 2021-04-23 at 1 13 25 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (adding or updating documentation)
- [ ] Dependency update

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -58,6 +59,7 @@ module.exports = {
     'react',
     'react-hooks',
     'you-dont-need-lodash-underscore',
+    'inclusive-language',
```
Comment (PixnBits) [inline index.js:62]: I _think_ this list is alphabetical?

**Review by PixnBits — CHANGES_REQUESTED** [review]
The comments on `unicorn.js` point out the usage of "master" in URLs
Looking I found https://github.com/muenzpraeger/eslint-plugin-inclusive-language/issues/1#issuecomment-657072516 which noted that "HEAD" could be used instead. IDK if all users have GitHub URLs but those are likely to be the bulk and others can add a `eslint-disable-next-line` if no other URL works

Might be good to update the URLs in unicorn.js in this PR

File: index.js (hunk 1 above)
Comment (smackfu) [reply index.js]: Fixed in c8c7ddafed3adcccf5e364e33139c933b894b6f9.

**Comment by smackfu** [comment]
> Might be good to update the URLs in unicorn.js in this PR

I fixed them in #70 by linking to tags, which actually is more correct since some of the links went 404 in the newer releases.

**Comment by nellyk** [comment]
@smackfu  since we are encouraging the use of inclusive language, i'm proposing we use `error` since resolving this would be a most likely a text change and most code bases are changing this. If someone isn't aware they might ignore the warning and reuse them again

**Review by infoxicator — DISMISSED** [review]

**Review by infoxicator — COMMENTED** [review]
I think going for error instead of warn is also fine, people tent to be more proactive when it stops what they are doing rather than a warning that could be missed

**Comment by smackfu** [comment]
@infoxicator @nellyk I changed it to be an "error" instead of "warning".  Thanks for the feedback!

**Review by nellyk — APPROVED** [review]

**Review by infoxicator — APPROVED** [review]

File: index.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable inclusive-language/use-inclusive-words */
```
Comment (PixnBits) [inline index.js:1]: I'd prefer this be disabled just for the rule configuration section (as I'm guessing that's where there are errors) but we also want to get this feature out

**Review by PixnBits — APPROVED** [review]

File: index.js (hunk 2 above)
Comment (smackfu) [reply index.js:1]: Fair point, I just figured all the lines in that section would need the exclusion which might make it less clear.

**Pull request merged**